rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java

```diff
@@ -0,0 +1,100 @@
+     * information which may be converted into JSON Object.
+     */
+    String uploadExternalResource(String path, String url);
+
+    /**
+     * A downloadBinFile method implementation is responsible for downloading a file-chunk from
+     * a cloud storage to user's local storage.
+     * @param path
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String downloadResource(String path);
+
+    /**
+     * Provides a copying procedure. Thus as a result one should have to similar files.
+     * @param from Stands for the path to a file that would be copied.
+     * @param to This is a new path for the copied file.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String copyFile(String from, String to);
+
+    /**
+     * This method implementation moves the file from one path to another.
+     * @param from Stands for the path to a file that would be copied.
+     * @param to This is a new path where the file will be moved.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String moveFile(String from, String to);
+
+    /**
+     * A createDirectory implementation method allows one to create a new folder or directory
+     * at a cloud storage.
+     * @param path A path where a new directory will be created relatively a root of a cloud storage.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String createDirectory(String path);
+
+    /**
+     * This method implementation should delete a directory or a file.
+     * @param path A path to a file which would be removed.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String deleteResource(String path);
+
+    /**
+     * A method implementation should provide the possibility to receive
+     * different information about a resource.
+     * E.g. creation time, file type, size and so on.
+     * Notice, that "resource" both means a usual file and a directory.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String getResourceMetadata(String path);
+
+
+    /**
+     * An implementation should allow one to get as much informaation
+     * about user's cloud disk storage as possible.
+     * Depending on a cloud storage provider the response may vary.
+     * At least the returned value should contain:
+     * - total space available
+     * - used space
+     * = free space
+     *
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String getCloudDiskMetadata();
```

Comment:
And in this interface we should be have methods for connect\disconnect, authenticate on cloud.